src/Lib/Canvas.vala

```diff
@@ -387,4 +390,92 @@ public class Akira.Lib.Canvas : Goo.Canvas {
         var cursor = new Gdk.Cursor.for_display (Gdk.Display.get_default (), cursor_type);
         get_window ().set_cursor (cursor);
     }
+
+    /**
+     * Handle the aftermath of an item transformation, like size changes or movement.
+     */
+    private void on_hold_release () {
+        if (selected_bound_manager.selected_items.length () == 0) {
+            return;
+        }
+
+        // Interrupt if no artboard is currently present.
+        if (window.items_manager.artboards.get_n_items () == 0) {
+            return;
+        }
+
+        // Check if any of the currently moved items was dropped inside or outside any artboard.
+        var items = selected_bound_manager.selected_items.copy ();
+        foreach (var item in items) {
+            if (item is Models.CanvasArtboard) {
+                continue;
+            }
+
+            foreach (Models.CanvasArtboard artboard in window.items_manager.artboards) {
+                var new_artboard = artboard.dropped_inside (item) ? artboard : null;
+                change_artboard (item, new_artboard);
+            }
+        }
+    }
+
+    /**
+     * Add or remove an item from an artboard.
+     */
+    public void change_artboard (Models.CanvasItem item, Models.CanvasArtboard? new_artboard) {
```

Comment:
Maybe we can move this logic out to the `ItemsManager` or `SelectedBoundManager` just to let the Canvas to deal _only_ with Canvas issues and business.

What do you think?